Deliverables/D2_project_decomposition.md

```diff
@@ -1,31 +1,35 @@
-3. A phone - The user will download the app, and they will need to open it to interact with it. Upon opening it, they will find... 
+2. The dataframe - The dataframe will hold the recipes and the information about them. Yield size, ingredients, instructions... This will all be organized into a data frame that can be parsed with the user's input. 
+3. .pkl file - This will be created after the webscraper collects data on the recipes. Once the dataframe is created, it needs to be "pickled", which ensures that my dataframe isn't just lost to the void. 
+
+~~3. A phone - The user will download the app, and they will need to open it to interact with it. Upon opening it, they will find...~~
 4. The Front-End UI - This is what the user will personally interact with to input their ingredients via a sort of text box search bar. After their ingredients are input, they hit go. This is when the background actions happen: 
-5. The Back-end Script: This will compare the user's input against what's in the dataframe, particularly the ingredients. A carefully constructed dataframe will be able to filter more user specifications later on. This will calculate a percentage match, which will then be displayed in...
+5. The Back-end Script: This will compare the user's input against what's in the ~~dataframe~~ .pkl file, particularly the ingredients. A carefully constructed dataframe will be able to filter more user specifications later on. This will calculate a percentage match, which will then be displayed in...
 6. Front-End UI Part 2: This is the page that will pop up after everything has been calculated, and will display the recipes for the user. It will display them in a scaling order based on the percentages matched, and display images of the recipe along with the name. The user will be able to click on the recipe and check it out, and back out to the results again if they so choose. 
 
 **What will the components need?**
 
 1. The webscraper I plan on using is located here: https://github.com/hhursev/recipe-scrapers
-2. The dataframe I've been researching is the one included within the Pandas package. 
+2. The dataframe I've been researching is the one included within the Pandas package. .pkl is also part of the Pandas package. 
 3. I intend on using Android Studio to create an environment to test the app in in place of the phone. 
-4. The front end UI I've been looking at TKinter for-- While I don't love how most of the stuff in it comes out, I think it's viable for what I plan. 
+4. The front end UI I've been looking at ~~TKinter for-- While I don't love how most of the stuff in it comes out, I think it's viable for what I plan.~~ Tkinter doesn't work on Android! What an oversight. I want to use kivy, which will work on Android. 
```

Comment:
Kivy should definitely be the way to go for this. Good decision. Here's an introductory tutorial for Kivy in case you wanna start learning it https://realpython.com/mobile-app-kivy-python/